### ISSUE TITLE
Update nginx ingress controller example for websockets

### ISSUE THREAD

**Issue by colearendt** [rstudio/helm#67]
After discussion with a customer, I am suspicious that websockets do not work with our nginx ingress controller (and that instead we are falling back to xhr-streaming). This is less than ideal because it impacts performance and prevents `shinyloadtest` from working.

We should update the example. Unfortunately, our need for a `map` [from our guide](https://docs.rstudio.com/connect/admin/proxy/#simple-configuration) is not possible with a generic `ingress` annotation, as far as I can tell.

https://docs.nginx.com/nginx-ingress-controller/configuration/ingress-resources/advanced-configuration-with-snippets/
https://docs.nginx.com/nginx-ingress-controller/configuration/ingress-resources/advanced-configuration-with-annotations/#snippets-and-custom-templates

More discussion:
- https://gist.github.com/jsdevtom/7045c03c021ce46b08cb3f41db0d76da#file-ingress-service-yaml
- https://stackoverflow.com/questions/52883653/nginx-ingress-controller-websocket-support#52884066

FYI @dbkegley

**Comment by colearendt** [comment]
Yeah, FWIW we are using traefik in those environments IIRC. The nginx ingress controller example is one @dbkegley wrote up, and I think nginx is not as friendly with websockets by default

**Comment by dbkegley** [comment]
Not surprising, nginx just isn't as friendly... I'll try to find some time to test this out but may not get to it until late next week. Based on the stackoverflow examples it sounds like they may have accomplished this just by enabling these annotations? Allegedly websockets should be enabled by default in nginx.

```
nginx.ingress.kubernetes.io/proxy-read-timeout: 3600
nginx.ingress.kubernetes.io/proxy-send-timeout: 3600
```

**Comment by colearendt** [comment]
I forget exactly what we did, but I remember we needed to add this to the main config block (not possible with annotations on the ingress). I believe we accomplished with an annotation on the ingress controller itself:
```
  map $http_upgrade $connection_upgrade {
    default upgrade;
    ''      close;
  }
```

And then this to the location block (via annotation):
```
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection $connection_upgrade;
```

**Comment by dbkegley** [comment]
We haven't seen any reports of this being an issue in quite some time. Please feel free to re-open if it is still a problem
